doc/design/diag_attribute_error.md

```diff
@@ -0,0 +1,96 @@
+# Diagnostic Attribute Error
+
+## Motivation
+
+SQLFlow extended SQL syntax to describe an end-to-end machine learning job, for a typical SQL program:
+
+``` python
+SELECT * FROM iris.train
+TO TRAIN DNNClassifier
+WITH
+  model.n_classes = 2,
+  model.hidden_units = [128, 64]
+LABEL class
+INTO my_model;
+
+SELECT * FROM iris.test
+TO PREDICT iris.pred.class
+USING my_model;
+```
+
+SQLFlow compiles the above SQL program into an execution plan and runs it.  As the TRAIN statement above, SQLFlow uses `TO TRAIN CLAUSE`  to train a specific model called  `DNNClassifier`, using `WITH CLAUSE` to configure the training arguments. 
+
+Sometimes users may make some configuration mistake on `WITH CLAUSE`, then the job would fault during execution and return some uncertain error message.
+
+This documentation issued a way that adding extra contract in [docstring](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) to contract the arguments, and this can achieve three advantage at least:
```

Comment:
Explain what contract is?

Reply:
We don't need the word contract here.  We can say 

The model parameter documentation describes parameters and acceptable values of human reading.  We want to enhance it for the reading by the SQLFlow compiler so to warn about wrongly set parameters.